crates/unftp-sbe-gcs/src/lib.rs

```diff
@@ -134,30 +138,91 @@ impl CloudStorage {
     {
         let client: Client<HttpsConnector<HttpConnector<GaiResolver>>, Body> =
             Client::builder().build(HttpsConnectorBuilder::new().with_native_roots().https_or_http().enable_http1().build());
-        CloudStorage {
+        Self {
             client,
             auth: auth.into(),
             uris: GcsUri::new(base_url.into(), bucket.into(), root),
+
+            cached_token: Default::default(),
         }
     }
 
-    // TODO: Cache the token. For `ServiceAccountKey`, the oauth client would already cache the token - we just need to move it to `CloudStorage`. For `WorkloadIdentity`, we can cache it in `CloudStorage`.
     #[tracing_attributes::instrument]
-    async fn get_token(&self) -> Result<String, Error> {
+    async fn get_token_value(&self) -> Result<String, Error> {
+        {
+            let cache = self.cached_token.read().unwrap();
```

Comment:
Since you now use the async lock you have to `await` it

Reply:
Also don't need to unwrap then I think cause the async one doesn't have the poisoning thing

Reply:
🤦  Forgot to cargo check before committing. Done